examples/api/lib/widgets/nested_scroll_view/nested_scroll_view.3.dart

```diff
@@ -0,0 +1,197 @@
+              // scroll view thinks it has not been scrolled.
+              // This is not necessary if the "headerSliverBuilder" only builds
+              // widgets that do not overlap the next sliver.
+              handle: NestedScrollView.sliverOverlapAbsorberHandleFor(context),
+              sliver: SliverAppBar(
+                title: const Text('Books'), // This is the title in the app bar.
+                pinned: true,
+                expandedHeight: 150.0,
+                // The "forceElevated" property causes the SliverAppBar to show
+                // a shadow. The "innerBoxIsScrolled" parameter is true when the
+                // inner scroll view is scrolled beyond its "zero" point, i.e.
+                // when it appears to be scrolled below the SliverAppBar.
+                // Without this, there are cases where the shadow would appear
+                // or not appear inappropriately, because the SliverAppBar is
+                // not actually aware of the precise position of the inner
+                // scroll views.
+                forceElevated: innerBoxIsScrolled,
+                bottom: TabBar(
+                  controller: _tabController,
+                  // These are the widgets to put in each tab in the tab bar.
+                  tabs: <Widget>[
+                    for (final String name in _tabs) Tab(text: name),
+                  ],
+                ),
+              ),
+            ),
+          ];
+        },
+        body: TabBarView(
+          controller: _tabController,
+          // These are the contents of the tab views, below the tabs.
+          children: <Widget>[
+            for (final (int index, String name) in _tabs.indexed)
+              TabViewContentExample(
+                name: name,
+                isSelected: index == _selectedIndex,
+              ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class TabViewContentExample extends StatefulWidget {
+  const TabViewContentExample({
+    required this.name,
+    required this.isSelected,
+    super.key,
+  });
+
+  final String name;
+  final bool isSelected;
+
+  @override
+  State<TabViewContentExample> createState() => TabViewContentExampleState();
+}
+
+class TabViewContentExampleState extends State<TabViewContentExample>
+    with AutomaticKeepAliveClientMixin {
+  @override
+  bool get wantKeepAlive => true;
+
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void dispose() {
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+
+    return SafeArea(
+      top: false,
+      bottom: false,
+      child: Builder(
+        // This Builder is needed to provide a BuildContext that is "inside"
+        // the NestedScrollView, so that sliverOverlapAbsorberHandleFor() can
+        // find the NestedScrollView.
+        builder: (BuildContext context) {
```

Comment:
I'm wondering if this `Builder` is necessary; it seems like `PrimaryScrollController.of(context)` would work using the state's build context.

Reply:
You're right. I removed the Builder and confirmed that everything still works fine without it.